packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1216,6 +1219,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       _currentAutofillScope?.unregister(autofillId);
       _currentAutofillScope = newAutofillGroup;
       newAutofillGroup?.register(this);
+      _isInAutofillContext = _isInAutofillContext || _shouldBeInAutofillContext;
```

Comment:
_isInAutofillContext can never be set to false once it becomes true, is that right?

Reply:
Yes. If a text field's was in the current autofill context, and then its autofill gets disabled (by setting `autofillHints` to null), the engine needs to remove it from the autofill context. It lets the engine know by sending its new `TextInputConfiguration`, instead of remain silent in which case the engine will just assume the text field is gone and it will keep the old user input.